src/hakunapi-jsonfg/src/main/java/fi/nls/hakunapi/geojson/hakuna/JSONFGFeatureCollectionWriter.java

```diff
@@ -0,0 +1,233 @@
+import fi.nls.hakunapi.core.projection.NOPProjectionTransformer;
+import fi.nls.hakunapi.core.projection.ProjectionHelper;
+import fi.nls.hakunapi.core.projection.ProjectionTransformer;
+import fi.nls.hakunapi.core.property.simple.HakunaPropertyGeometry;
+import fi.nls.hakunapi.core.schemas.Crs;
+import fi.nls.hakunapi.core.schemas.Link;
+import fi.nls.hakunapi.core.util.CrsUtil;
+import fi.nls.hakunapi.jsonfg.JSONFG;
+
+public class JSONFGFeatureCollectionWriter extends HakunaGeoJSONFeatureCollectionWriter implements JSONFG {
+
+	protected int outputSrid;
+	protected JSONFGGeometryWriter placeJson;
+
+	// local cached (+inherited geometryCached & cachedGeometry )
+	FeatureType collectionFt = null;
+	LocalDate date = null;
+	Instant timestamp = null;
+	ProjectionTransformer outputWgs84Proj;
+
+	@Override
+	public void init(OutputStream out, FloatingPointFormatter formatter, int srid, boolean crsIsLatLon)
+			throws IOException {
+		super.init(out, formatter, Crs.CRS84_SRID, crsIsLatLon);
+		outputSrid = srid;
+
+	}
+
+	@Override
+	public void initGeometryWriter(HakunaGeometryDimension dims) {
+		this.dims = dims;
+		geometryJson = new JSONFGGeometryWriter(json, HakunaGeoJSON.GEOMETRY, true, dims);
+		placeJson = new JSONFGGeometryWriter(json, PLACE, true, dims);
+		propertyGeometryJson.clear();
+	}
+
+	protected void writeFeatureCollection() throws IOException {
+	}
+
+	@Override
+	public void writeProperty(String name, HakunaGeometry geometry) throws Exception {
+		openProperties();
+		final GeometryWriter geomWriter = propertyGeometryJson.computeIfAbsent(name, k -> new JSONFGGeometryWriter(json,
+				k.getBytes(StandardCharsets.UTF_8), forceLonLat || !crsIsLatLon, dims));
+		geometry.write(geomWriter);
+	}
+
+	@Override
+	public void startFeatureCollection(FeatureType ft, String layername) throws Exception {
+
+		collectionFt = ft;
+		if (isWGS84(outputSrid)) {
+			outputWgs84Proj = NOPProjectionTransformer.INSTANCE;
+		} else {
+			outputWgs84Proj = collectionFt.getProjectionTransformerFactory().toCRS84(outputSrid);
+		}
+
+		boolean isWgs84 = isWGS84(outputSrid);
+		String ftName = ft.getName();
+		String ftJSONFGSchema = getJSONFGSchema(ft);
+
+		HakunaPropertyGeometry geomType = ft.getGeom();
+		Integer geometryDimension = getGeometryDimension(geomType);
+
+		json.writeStartObject();
+
+		// conformsTo
+		json.writeFieldName(CONFORMS_TO);
+		json.writeStartArray();
+		for (byte[] conf : CONF) {
+			json.writeStringUnsafe(conf);
+		}
+		json.writeEndArray();
+
+		// type
+		json.writeFieldName(HakunaGeoJSON.TYPE);
+		json.writeStringUnsafe(HakunaGeoJSON.FEATURE_COLLECTION, 0, HakunaGeoJSON.FEATURE_COLLECTION.length);
+
+		json.writeStringField(FEATURE_TYPE, ftName);
+
+		if (geometryDimension != null) {
+			json.writeFieldName(GEOMETRY_DIMENSION);
+			json.writeNumber(geometryDimension);
+		}
+		if (ftJSONFGSchema != null) {
+			json.writeStringField(FEATURE_SCHEMA, ftJSONFGSchema);
+		}
+
+		if (!isWgs84) {
```

Comment:
If we let ProjectionTransformerFactory determine if the outputSrid is WGS84 compliant then we could use `outputWgs84Proj.isNOP()` to determine if `coordRefSys` needs to be written out.

Reply:
retained srid==84 check for now NOP check did not work